src/logic/prices/marketplace.ts

```diff
@@ -96,13 +104,18 @@ export function marketplacePricesQuery(filters: PriceFilters) {
     searchEstateSize_gt: 0,
     ${additionalWheres.join('\n')}
   `
-  return `query NFTPrices(
-      $expiresAt: String
-      $expiresAtSec: String
-      $wearableCategory: String
-      $isWearableHead: Boolean
-      $isWearableAccessory: Boolean
-      ) {
+
+  const variablesBasedOnFilters = Object.entries(filters)
+    .filter(([key, value]) => value !== undefined && key in PRICES_FILTERS_DICT)
+    .map(([key]) => (PRICES_FILTERS_DICT[key] ? PRICES_FILTERS_DICT[key] : ''))
```

Comment:
What do you think about abstracting this mechanism so we can use it in all of these queries?

Reply:
Sure, great idea, let me prepare that in a following up PR so we can review that exclusively.